Added some highlights for .NET MAUI (#43411)
dotnet/docs: docs/core/whats-new/dotnet-9/overview.md

## .NET MAUI

> The focus of .NET Multi-platform App UI (.NET MAUI) in .NET 9 is to improve product quality. In addition:

Reviewer: ```suggestion
The focus of .NET Multi-platform App UI (.NET MAUI) in .NET 9 is focused on enhanced performance, reliability, and deeper integrations for desktop and mobile applications. .NET MAUI includes a new more performant implementation of `CollectionView` & `CarouselView` for iOS & Mac Catalyst, updates to existing controls, new app lifecycle events, and Native AOT and trimming enhancements to improve app size and startup time.  In addition:
```

> The focus of .NET Multi-platform App UI (.NET MAUI) in .NET 9 is enhanced performance, reliability, and deeper integrations for desktop and mobile applications. .NET MAUI includes a new more performant implementation of `CollectionView` & `CarouselView` for iOS & Mac Catalyst, updates to existing controls, new app lifecycle events, and Native AOT and trimming enhancements to improve app size and startup time.  In addition:

Reviewer 2: ```suggestion
The focus of .NET Multi-platform App UI (.NET MAUI) in .NET 9 is enhanced performance and reliability, and deeper integrations for desktop and mobile applications. .NET MAUI includes a new, more performant implementation of <xref:Microsoft.Maui.Controls.CollectionView> and <xref:Microsoft.Maui.Controls.CarouselView> for iOS and Mac Catalyst, updates to existing controls, new app lifecycle events, and Native AOT and trimming enhancements to improve app size and startup time.  In addition:
```

> - A new `TitleBar` desktop control is available for Windows
> - A new `HybridWebView` control enables easier inclusion of JavaScript enabled content from frameworks like ReactJS, Vue.js, and Angular
> - `Entry` now supports additional keyboard modes
> - Control handlers automatically disconnect from their controls when possible
> - `MainPage` is deprecated in favor of setting the primary page of the app by overriding `CreateWindow` in the `App` class

Reviewer 2: ```suggestion
- A new <xref:Microsoft.Maui.Controls.TitleBar> desktop control is available for Windows.
- A new <xref:Microsoft.Maui.Controls.HybridWebView> control enables easier inclusion of JavaScript-enabled content from frameworks like ReactJS, Vue.js, and Angular.
- <xref:Microsoft.Maui.Controls.Entry> now supports additional keyboard modes.
- Control handlers automatically disconnect from their controls when possible.
- <xref:Microsoft.Maui.Controls.Application.MainPage> is deprecated in favor of setting the primary page of the app by overriding <xref:Microsoft.Maui.Controls.Application.CreateWindow(Microsoft.Maui.IActivationState)?displayProperty=nameWithType> class.
```

> For more information about that and new features, see [What's new in .NET MAUI for .NET 9](/dotnet/maui/whats-new/dotnet-9).

Reviewer: ```suggestion
For more information about that these new features and more, see [What's new in .NET MAUI for .NET 9](/dotnet/maui/whats-new/dotnet-9).
```
